src/views/events.js

```diff
@@ -141,6 +141,16 @@ var EventsView = JuttleView.extend({
         this.target.stream_end();
     },
 
+    _validateTimeField : function(time) {
+        // if in timechart overlay, timeField needs to actually be a Date
+        if (this._targetView) {
+            return _.isDate(time) && !_.isNaN(time.getTime());
```

Comment:
The `_.isNumber` half of what they do is useless and slow as molasses. If performance isn't important though `_.isNaN` has the readability edge.